packages/functional-tests/tests/misc/recoveryKeyStretchingV2.spec.ts

```diff
@@ -0,0 +1,156 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { EmailHeader, EmailType } from '../../lib/email';
+import { expect, test } from '../../lib/fixtures/standard';
+import { BaseTarget } from '../../lib/targets/base';
+
+const AGE_21 = '21';
+const HINT = 'secret key location';
+
+/**
+ * These tests represent various permutations between interacting with V1 and V2
+ * key stretched passwords. We need to ensure that operations are interchangeable!
+ */
+test.describe('severity-2 #smoke', () => {
+  test.describe('key-stretching-v2', () => {
+    test.slow();
+
+    // Helpers
+    async function _getKeys(
+      version: 1 | 2,
+      target: BaseTarget,
+      email: string,
+      password: string
+    ) {
+      const client = target.createAuthClient(version);
+      const response = await client.signIn(email, password, { keys: true });
+      const keys = client.accountKeys(
+        response.keyFetchToken,
+        response.unwrapBKey
+      );
+      return keys;
+    }
```

Comment:
It's also repeated in resetPasswordKeyStretchingV2.spec.ts unfortunately. I think there is opportunity to make it DRYer ... probably by moving this logic into Target...? I'm not confident enough that this would improve things atm so will opt for the 'later' 😬